.github/workflows/validate.yml

```diff
@@ -83,3 +83,50 @@ jobs:
           flags: tests
           name: codecov-umbrella
           verbose: true
+
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        id: setup-python
+        with:
+          python-version: "3.10"
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@d45b6d76012debf457ab49dffc7fb7b2efe8071d
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+      #----------------------------------------------
+      # install dependencies if cache does not exist
+      #----------------------------------------------
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+      #----------------------------------------------
+      # install your root project, if required
+      #----------------------------------------------
+      - name: Install library
+        run: poetry install --no-interaction
+      #----------------------------------------------
+      # run 'black', will use installed version
+      #----------------------------------------------
```

Comment:
Could probably abstract this all to a composite action in the future.

Reply:
I'm quite the GitHub Action novice but if we can abstract this duplicate behavior into one place that would be great.

Reply:
This is something we could do in this PR or after as a cleanup. @nemith would you be willing to follow up with one? If not please file an issue so we do not forget it.

@jtwb here is documentation on composite actions
https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

Reply:
This was resolved in https://github.com/aerleon/aerleon/pull/197